### PR TITLE
[Prot Pally] Changed Hammer of the Righteous to track non-filler casts

### PR DESCRIPTION
**Pull request by emallson**
Hammer of the Righteous is a filler spell, doing almost no damage relative to Judgment and Avenger's Shield. It is *always* correct to let it sit at two charges when you have either of the others to cast, including when you switch to the Blessed Hammer talent (which replaces HotR).

This changes the HotR/BH checklist item and suggestion from tracking cast efficiency to tracking bad (non-filler) casts. As far as I know, there isn't a standard set of tools for doing this so I implemented it manually.

I chose to use `isOnCooldown` rather than `isAvailable` deliberately, because it can be correct to use HotR over your second charge of Judgment to put HotR on CD when running the Crusader's Judgment talent (which gives Judgment 2 charges).

screenshot from 2018-11-25 15-50-06
screenshot from 2018-11-25 15-49-54

### CODE REVIEW

File: src/parser/paladin/protection/modules/features/Checklist/Module.js [hunk 1]
```diff
@@ -30,13 +32,15 @@ class Checklist extends BaseChecklist {
         castEfficiency={this.castEfficiency}
         extras={{
           lotpAbility: this.lotp._activeSpell,
+          hotrAbility: this.hotr._activeSpell,
```
Comment (Dambroda) [inline src/parser/paladin/protection/modules/features/Checklist/Module.js]: My understanding was that the underscore variable prefix basically meant that it shouldn't be accessed from outside the class -- is this intentional?

File: src/parser/paladin/protection/modules/features/Checklist/Component.js [hunk 2]
```diff
@@ -39,8 +39,13 @@ class ProtectionPaladinChecklist extends React.PureComponent{
           <AbilityRequirement spell={SPELLS.SHIELD_OF_THE_RIGHTEOUS.id} />
           <AbilityRequirement spell={SPELLS.AVENGERS_SHIELD.id} />
           <AbilityRequirement spell={SPELLS.JUDGMENT_CAST_PROTECTION.id} />
-          {combatant.hasTalent(SPELLS.BLESSED_HAMMER_TALENT.id) && <AbilityRequirement spell={SPELLS.BLESSED_HAMMER_TALENT.id} />}
-          {!combatant.hasTalent(SPELLS.BLESSED_HAMMER_TALENT.id) && <AbilityRequirement spell={SPELLS.HAMMER_OF_THE_RIGHTEOUS.id} />}
+          <Requirement
+            name={(
```
Comment (Dambroda) [inline src/parser/paladin/protection/modules/features/Checklist/Component.js]: Parenthesis could be removed and fragment on one line; but I am not sure which style is preferred

File: src/parser/paladin/protection/modules/features/Checklist/Module.js (hunk 1 above)
Comment (emallson) [reply src/parser/paladin/protection/modules/features/Checklist/Module.js]: you're right, i'll fix these.

**Comment by emallson** [comment]
@Dambroda any more comments?

**Pull request merged**